packages/core/src/consumed-thing.ts

```diff
@@ -450,19 +450,19 @@ export default class ConsumedThing extends TD.Thing implements IConsumedThing {
         for (const s of security) {
             const ws = this.securityDefinitions[s + ""]; // String vs. string (fix wot-typescript-definitions?)
             // also push nosec in case of proxy
-            if (ws) {
+            if (ws != null) {
                 scs.push(ws);
             }
         }
         return scs;
     }
 
     ensureClientSecurity(client: ProtocolClient, form: TD.Form | undefined): void {
-        if (this.securityDefinitions) {
+        if (this.securityDefinitions != null) {
             const logStatement = () =>
                 debug(`ConsumedThing '${this.title}' setting credentials for ${client} based on thing security`);
 
-            if (form && Array.isArray(form.security) && form.security.length > 0) {
+            if (form != null && Array.isArray(form.security) && form.security.length > 0) {
```

Comment:
I think this if block can be further simplified if a `security` auxiliary variable is introduced like so:

```suggestion
            const security = form?.security;
            if (Array.isArray(security) && security.length > 0) {
```

Reply:
In this case I tend to stick to the existing form ...